cpp/ql/src/Likely Bugs/Memory Management/UninitializedLocal.ql

```diff
@@ -40,23 +40,17 @@ DeclStmt declWithNoInit(LocalVariable v) {
   result.getADeclaration() = v and
   not exists(v.getInitializer()) and
   /* The type of the variable is not stack-allocated. */
-  not allocatedType(v.getType()) and
-  /* The variable is not static (otherwise it is zeroed). */
-  not v.isStatic() and
-  /* The variable is not extern (otherwise it is zeroed). */
-  not v.hasSpecifier("extern")
```

Comment:
Was this line doing anything?  (your test case seems to suggest not)

Reply:
I believe that line had no effect, and I added the test case as proof.